src/types.rs

```diff
@@ -103,12 +104,13 @@ pub(crate) fn least_upper_bound(mut tags: impl Iterator<Item = TypeBound>) -> Ty
     .into_inner()
 }
 
+#[allow(missing_docs)]
 #[derive(Clone, PartialEq, Debug, Eq, derive_more::Display, Serialize, Deserialize)]
 #[serde(tag = "s")]
 /// Representation of a Sum type.
 /// Either store the types of the variants, or in the special (but common) case
 /// of a "simple predicate" (sum over empty tuples), store only the size of the predicate.
```

Comment:
i think this would allow no docs on the enum itself, maybe instead remove the #[allow] and split the docstring up between the variants (it currently describes both of them at the top)

Reply:
I like the single comment approach and so have moved the #[allow ] to the variants. Will change if you prefer.
I applied the same change to TypeEnum.